### PR TITLE
Install onlyoffice including dependecies apps

### PR DESCRIPTION
**Pull request by Edmondi-Kacaj**
Extend ansible for providing a way to preview Microsoft office files, like .doc(x), .ppt(x), .xls(x).

Changes consist in : 
  -  installing and configuring  Nginx-extra
  -  installing and configuring  RabbitMQ  
  -  installing and configuring  PostgreSQL
  -  installing and configuring  only office document server (community edition)
  - possibility to add (override) config files for render-service app

### CODE REVIEW

**Review by mirjan-hoffmann — CHANGES_REQUESTED** [review]
I've had a rough look over it and I've noticed a few points / questions that should still be adjusted / answered:

- The installation of the OnlyOffice components cannot be enabled/disabled via Ansible variable -> this still needs to be built in and should default to "false".
- There is a separate group set up for OnlyOffice and OnlyOffice is independent from the existing components -> therefore should be installed only on OnlyOffice hosts and not always on the renderingservice hosts (of course this can be done via Config)
- If installed on one of the existing hosts, then there is already a web server (apache2). Why install a second one (nginx)? Better use the existing apache2 and create additional apache2-config!
- Include only the most necessary configuration in all.yml. If possible, everything that concerns onlyoffice should also be included in the corresponding onlyoffice.yml.
- Are SSL certificates mandatory? So far we do the encryption of the other connections in the public systems via the ReverseProxy....
- Typo: "postgress" -> should be "postgres"
- A galaxy-role like https://galaxy.ansible.com/geerlingguy/rabbitmq is not sufficient to install rabbitmq? Do we really need to implement this role ourself?

**Comment by Edmondi-Kacaj** [comment]
Hello @mirjan-hoffmann , I made all those necessary changes, can you check again 

best regard
Edmond

**Comment by mirjan-hoffmann** [comment]
@Edmondi-Kacaj ok, good, most of the points mentioned above are implemented now. But some are not (I know we talked directly about some of them and why they should not be implemented). In order to be able to understand this implementation later, please document here why you could not implement it:

> If installed on one of the existing hosts, then there is already a web server (apache2). Why install a second one (nginx)? Better use the existing apache2 and create additional apache2-config!

-> besides the description why this is not implemented, maybe also the link to the official documentation concerning nginx

> Include only the most necessary configuration in all.yml. If possible, everything that concerns onlyoffice should also be included in the corresponding onlyoffice.yml.

-> I know nginx and rabbitmq could be installed independently from onlyoffice, but atm we only need it for onlyoffice. So, I think it would be ok to include the coresponding variables in onlyoffice.yml. Then it is more clear that this configuration needs only to be adjusted in case someone uses onlyoffice. Or is this not possible?

Since this change is somewhat larger here, it would also be good to create a separate ONLYOFFICE.md in which the feature is documented (description, what is installed and why, what function does it provide, how is it configured,...) - could you please do that in addition?

**Comment by Edmondi-Kacaj** [comment]
Hello @mirjan-hoffmann 

I added the Documentation for onlyoffice, and also I remove rabbitMQ installation variable from all.yml to onlyoffice.yml, about the NGINX configuration, I just comment because we don't use them for onlyoffice since it comes with it own configuration.

Best regards
Edmond

**Comment by mirjan-hoffmann** [comment]
Thanks @Edmondi-Kacaj, I'll try to run it locally before I merge

Can you please explain here in the issue the open question from above: description why we will use nginx here instead of using the existing apache2

> > If installed on one of the existing hosts, then there is already a web server (apache2). Why install a second one (nginx)? Better use the existing apache2 and create additional apache2-config!

> -> besides the description why this is not implemented, maybe also the link to the official documentation concerning nginx

File: ansible/group_vars/renderingservice.yml [hunk 1]
```diff
@@ -20,4 +20,4 @@ esrender_exec_method: http
 esrender_archive_retrieve_method: download
 esrender_local_archive_path:
 
-install_splash: false # install splash on renderingservice-hosts - see https://splash.readthedocs.io/en/stable/index.html
+install_splash: false # install splash on renderingservice-hosts - see https://splash.readthedocs.io/en/stable/index.html
```
Comment (mirjan-hoffmann) [inline ansible/group_vars/renderingservice.yml:23]:
```suggestion
install_splash: false # install splash on renderingservice-hosts - see https://splash.readthedocs.io/en/stable/index.html

```

Comment (mirjan-hoffmann) [reply ansible/group_vars/renderingservice.yml:23]: we should not change the format of the file, if there is no functional change at all

**Review by mirjan-hoffmann — CHANGES_REQUESTED** [review]

**Comment by Edmondi-Kacaj** [comment]
Hello @mirjan-hoffmann 

I have documented in [OnlyOffice.md](https://github.com/TIBHannover/edu-sharing-box/pull/12/commits/c5000116fc61f5d1dc20d874c9a21e81a366039f)  why we need an Nginx webserver.

OnlyOffice is built on top of Nginx, and when we install it, it will provide all configurations for Nginx. and doesn't have a possibility to choose another webserver.  so for now the only way to run Onlyoffice is Nginx.

we can use apache as a reverse proxy, but I don't think we need it since it still needs Nginx to run OnlyOffice.

**Review by mirjan-hoffmann — APPROVED** [review]
+1

**Pull request merged**